synapse/handlers/room_member.py

```diff
@@ -566,7 +568,13 @@ async def copy_user_state_on_room_upgrade(self, old_room_id, new_room_id, user_i
                 )
                 continue
 
-    async def send_membership_event(self, requester, event, context, ratelimit=True):
+    async def send_membership_event(
+        self,
+        requester: Requester,
+        event: EventBase,
+        context: EventContext,
+        ratelimit: bool = True,
```

Comment:
You probably want to remove the types from the docstring of that function (especially as the types listed there seem outdated).